File: numerical_calculations/task1.py
import math
#            1- cos(x)
# lim 4038 * --------- = 2019
#              x^2
#
# De l'Hospital
# 
# 

def checkCOS():
    x = 1.0
    while math.cos(x) != 1.0:
        x /= 10
        print("cos(x) = ",math.cos(x),"\tx = ",x)

def f(x):
    return 4038 * (1-math.cos(x))/math.pow(x,2)

def task():
    x = 0.00000000001
    for i in range(11, 21):
        print("f(x) = ",f(x),"\tfor x = ",x)
        x/=10

print("experiment for task 1:")
task()
print("\n\nlet's check how cosinus works on 10^-i:\n")
checkCOS()

File: numerical_calculations/TASK1.cpp
#include <iostream>
#include<math.h>
#include <iomanip>

using namespace std;

// cos(10^-7) = 0.99999999999995
//cos(10^-i) = 1.0 gdzie i = 8, 9, 10 ...

float f(long double x){
    return 4038 * (1 - cos(x)) / (pow(x,2.0));
}

long double f2(long double x){
    return 4038 * (1 - cos(x)) / (pow(x,2.0));
}

int main()
{
    //for(int i=11; i<=20; i++){
    //    cout<<f2(pow(10,-i))<<endl;
    //}
    long double x = 0.01;
    while(cos(x)!=1){
        x /= 10.0;
        cout<<setprecision(10)<<(double)cos(x)<<"     x = "<<x<<endl;
    }
    return 0;
}
